internal/controller/drplacementcontrol_controller.go

```diff
@@ -65,6 +65,18 @@ var InitialWaitTimeForDRPCPlacementRule = errorswrapper.New("Waiting for DRPC Pl
 // ProgressCallback of function type
 type ProgressCallback func(string, string)
 
+// DeleteInProgress error is returned during DRPC deletion when we need to wait and try later.
+type deleteInProgress struct{ string }
+
+func (e deleteInProgress) Error() string { return e.string }
```

Comment:
This approach works and looks elegant, but why introduce the complexity of defining a new type? Why not simply use `errors.New()` [similar to these](https://github.com/RamenDR/ramen/blob/6c96132862edcc106bf37bd5dc951a003a9175d5/internal/controller/drplacementcontrol.go#L43) ? One might argue it’s for `errors.Is()` support, but is it necessary to check for this specific error type at line 189? Instead, consider returning a custom error directly. The reconciler will eventually retry, and the error will still be logged.

Reply:
> This approach works and looks elegant, but why introduce the complexity of defining a new type? Why not simply use `errors.New()` [similar to these](https://github.com/RamenDR/ramen/blob/6c96132862edcc106bf37bd5dc951a003a9175d5/internal/controller/drplacementcontrol.go#L43) ?

Using a sentinel error works when there one special error like io.EOF. But we have multiple reasons to requeue the request during delete:

- adopting a vrg
- waiting until secondary is deleted
- waiting until primary is deleted
- vrg count > 0 (maybe not possible now)

Using a a type matching any error of the same type, we can return multiple errors have the same handling.

We already use this in https://github.com/RamenDR/ramen/blob/6c96132862edcc106bf37bd5dc951a003a9175d5/internal/controller/kubeobjects/requests.go#L141

I started with a local error in this package, but this can move to a shared package and used in all controllers.

> One might argue it’s for `errors.Is()` support, but is it necessary to check for this specific error type at line 189? 

Yes, see comment bellow.

> Instead, consider returning a custom error directly. The reconciler will eventually retry, and the error will still be logged.

This is what the first commit is doing, using fmt.Errorf() - the result is having 18 errors with stacktrace for every disable dr. This is huge amount of noise and it is incorrect to log a normal condition as an error.

Reply:
See the errors generated by the first commit - each one has a 10 lines stacktrace that I did not copy:
https://github.com/RamenDR/ramen/pull/1689#issuecomment-2510227228

Reply:
Using an error is just a limit of the current code, that have single error return value for the entire finalizeDRPC code path. We can change this but this is a big change that will be hard to backport, and I don't think it will be better to return multiple arguments like we have in other places. Using specific error type for the common case of "not ready yet" seems better.

Reply:
> This approach works and looks elegant, but why introduce the complexity of defining a new type?

Using existing type now.